samples/XAMLCatalog/XAMLCatalog/UIViewViewController.m

```diff
@@ -0,0 +1,158 @@
+    _shouldRespondToTouch = shouldRespond;
+}
+
+@end
+
+@implementation UIViewViewController {
+    CGRect _labelFrame;
+    CGRect _containerViewFrame;
+    CGRect _testViewFrame;
+    CGRect _statusLabelViewFrame;
+    float _cellHeight;
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.title = @"UIView";
+    [self tableView].allowsSelection = NO;
+
+    static const float margin = 5.0;
+
+    _labelFrame = CGRectMake(margin, margin, 600, 50);
+    _containerViewFrame = CGRectMake(margin, margin + _labelFrame.size.height, 300, 100);
+    _testViewFrame = CGRectMake(margin, margin, _containerViewFrame.size.width - (margin * 2.0), _containerViewFrame.size.height - (margin * 2.0));
+    _statusLabelViewFrame = CGRectMake(margin * 2.0, margin, _testViewFrame.size.width - (margin * 2.0), 40);
+    _cellHeight = _labelFrame.size.height + _containerViewFrame.size.height + (margin * 2.0);
+}
+
+- (NSInteger)tableView:(UITableView*)tableView numberOfRowsInSection:(NSInteger)section {
+    return 5;
+}
+
+- (CGFloat)tableView:(UITableView*)tableView heightForRowAtIndexPath:(NSIndexPath*)indexPath {
+    return _cellHeight;
+}
+
+- (UITableViewCell*)tableView:(UITableView*)tableView cellForRowAtIndexPath:(NSIndexPath*)indexPath {
+    // Grab a cell
+    UITableViewCell* cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"MenuCell"];
+
+    // Add the instruction label
+    UILabel* label = [[UILabel alloc] initWithFrame:_labelFrame];
+    [cell.contentView addSubview:label];
+
+    // Add a container view to deliniate bounds of test view
+    UIView* containerView = [[UIView alloc] initWithFrame:_containerViewFrame];
+    containerView.backgroundColor = [UIColor grayColor];
+    [cell.contentView addSubview : containerView];
+
+    // Add the test view
+    UIViewHitTest* testView = [[UIViewHitTest alloc] initWithFrame:_testViewFrame];
+    testView.backgroundColor = [UIColor yellowColor];
+    [containerView addSubview:testView];
+
+    // Add the status label
+    UILabel* statusLabel = [[UILabel alloc] initWithFrame:_statusLabelViewFrame];
+    statusLabel.text = @"No Touch Input Detected";
+    statusLabel.userInteractionEnabled = NO;
+    [containerView addSubview:statusLabel];
+
+    // Give the testView a label to report status
+    testView.statusLabel = statusLabel;
+
+    if (indexPath.row == 0) {
+        // Setup test
+        label.text = @"Alpha=0.0; yellow background SHOULD NOT be visible and view SHOULD NOT respond to touch.";
+        testView.alpha = 0.0;
+        testView.shouldRespondToTouch = NO;
+    } else if (indexPath.row == 1) {
+        // Setup test
+        label.text = @"Yellow background SHOULD be visible and view SHOULD respond to touch.";
+    } else if (indexPath.row == 2) {
+        // Setup test
+        label.text = @"Clear background SHOULD NOT be visible, but view SHOULD respond to touch.";
+        testView.backgroundColor = [UIColor clearColor];
```

Comment:
It should be have the same as clear, but good idea.

---
In reply to: [95709193](https://github.com/Microsoft/WinObjC/pull/1685#discussion_r95709193) [](ancestors = 95709193)